src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,38 +28,129 @@ public class FindNewsCommand : IFindNewsCommand
     private readonly INewsRepository _repository;
     private readonly INewsInfoMapper _mapper;
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IRequestClient<IGetUsersDataRequest> _usersRequestClient;
+    private readonly IRequestClient<IGetDepartmentsRequest> _departmentsRequestClient;
+    private readonly ILogger<GetNewsCommand> _logger;
+
+    private List<UserData> GetAuthor(List<Guid> authorIds, List<string> errors)
+    {
+      if (authorIds == null || authorIds.Count == 0)
+      {
+        return null;
+      }
+
+      string errorMessage = "Cannot get authors now. Please try again later.";
+      const string logMessage = "Cannot get authors with ids: {authorIds}.";
+
+      try
+      {
+        object request = IGetUsersDataRequest.CreateObj(authorIds);
+        Response<IOperationResult<IGetUsersDataResponse>> response = _usersRequestClient
+          .GetResponse<IOperationResult<IGetUsersDataResponse>>(request)
+          .Result;
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.UsersData;
+        }
+
+        _logger.LogWarning(logMessage, string.Join(", ", authorIds));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, string.Join(", ", authorIds));
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
+
+    private List<DepartmentData> GetDepartments(List<Guid> departmentIds ,List<string> errors)
```

Comment:
same comments

Reply:
and fix style of arguments, the comma is incorrectly placed

Reply:
Why is there async?